### PR TITLE
Architecture

### PR DESCRIPTION
**Pull request by Logopeddd**
- File structure update
- Duks
- Filter update, sync filter with URL
- Axios integrated, requests added
- Small styles fix

### CODE REVIEW

File: src/screens/catalog.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  componentDidMount() {
+    this.props.loadProducts(this.state.values);
+  }
+
+  render() {
+    const { filterOnDisplay, products, match: { params } } = this.props;
+
+    let filter;
```
Comment (aaneitchik) [inline src/screens/catalog.js]: maybe it's just me, but if a condition for displaying a component is simple (like in this case), I prefer using it in render() like so: {filterOnDisplay && <FilterComponent/>}

File: src/screens/catalog.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const mapStateToProps = state => ({
+  filterOnDisplay: state.filter,
+  products: state.products,
+});
+
+CatalogPage.defaultProps = {
+  products: [],
+};
+
+CatalogPage.propTypes = {
```
Comment (aaneitchik) [inline src/screens/catalog.js]: there's a tip I learnt recently: when you have a lot of different PropTypes, you can do:
const { arrayOf, bool, func } = PropTypes
looks really nice :)

**Review by aaneitchik — APPROVED** [review]
I'm kind of proud :) looks really nice and clean, with prettier, ducks and propTypes it almost feels like I'm in my own codebase)

File: src/screens/catalog.js (hunk 1 above)
Comment (Logopeddd) [reply src/screens/catalog.js]: just I saw it in Grommet demoApp, but I agree that {filterOnDisplay && } look cleaner

File: src/screens/catalog.js (hunk 2 above)
Comment (Logopeddd) [reply src/screens/catalog.js]: lol, sure)

**Pull request merged**